docs/source/ext/binding-filter.rst

```diff
@@ -13,13 +13,68 @@ StreamFlow lets users map steps to :ref:`multiple targets <Multiple targets>`. A
 
 The ``get_targets`` method receives a ``Job`` object and a list of ``Target`` objects, the list of targets specified by the user, and returns another list of ``Target`` objects. The :ref:`Scheduler <Scheduler>` component will evaluate the returned list of targets to find an allocation for the ``Job`` object.
 
-By default, if no ``BindingFilter`` is specified for a multi-target step binding, all the ``Target`` objects will be evaluated in the original order. In addition, StreamFlow defines a ``ShuffleBindingFilter`` implementation to randomise the evaluation order at any invocation.
+By default, if no ``BindingFilter`` is specified for a multi-target step binding, all the ``Target`` objects will be evaluated in the original order.
 
 Implementations
 ===============
 
-=======     =========================================================
-Type        Class
-=======     =========================================================
-shuffle     streamflow.deployment.filter.shuffle.ShuffleBindingFilter
-=======     =========================================================
+=========     ==========================================================
+Type          Class
+=========     ==========================================================
+shuffle       streamflow.deployment.filter.shuffle.ShuffleBindingFilter
+matching      streamflow.deployment.filter.shuffle.MatchingBindingFilter
+=========     ==========================================================
+
+
+ShuffleBindingFilter
+^^^^^^^^^^^^^^^^^^^^
+The ``ShuffleBindingFilter`` implementation does not discard any location, but simply randomizes the evaluation order at each invocation.
+
+MatchingBindingFilter
+^^^^^^^^^^^^^^^^^^^^^
+The ``MatchingBindingFilter`` filters ``target`` objects by applying an exact ``match`` between an expected value and job input ``ports``. A ``deployment`` is considered a match if its name matches one of the ``targets`` and the input values of the ``Job`` object satisfies the specified value. Otherwise, the ``target`` is discarded.
```

Comment:
```suggestion
The ``MatchingBindingFilter`` filters ``target`` objects by applying an exact ``match`` between an expected value and job input ``ports``. A ``deployment`` is considered a match if its name matches one of the ``targets`` and the input values of the ``Job`` object satisfy the specified value. Otherwise, the ``target`` is discarded.
```